addons/mail/static/src/core/common/store_service.js

```diff
@@ -554,30 +567,31 @@ export class Store extends BaseStore {
         }
         if (partner_ids.length) {
             Object.assign(postData, { partner_ids });
+            this.fillPartnersMentionToken(postData);
         }
         if (role_ids.length) {
             Object.assign(postData, { role_ids });
         }
         if (thread.model === "discuss.channel" && validMentions?.specialMentions.length) {
             postData.special_mentions = validMentions.specialMentions;
         }
+        if (attachments.length) {
+            postData.attachment_tokens = attachments.map(
+                (attachment) => attachment.ownership_token
+            );
+        }
+        if (recipientEmails.length) {
+            postData.partner_emails = recipientEmails;
+        }
```

Comment:
Is this really that bad to send an empty array? It's conceptually better IMO and it would ease those conditions, making the method easiere to follow.